edge-modules/edgehub-proxy/src/error.rs

```diff
@@ -1,113 +1,59 @@
 // Copyright (c) Microsoft. All rights reserved.
 
-use std::fmt::{self, Display};
+//use std::fmt::{self, Display};
 use std::io;
 
-use failure::{Backtrace, Context, Fail};
+use thiserror::Error as ThisError;
 use hyper::Error as HyperError;
 use serde_json;
-use url::ParseError;
+use chrono::ParseError;
+use edgelet_client::Error as ClientError;
+use edgelet_client::WorkloadError;
 
-use edgelet_http::Error as EdgeletHttpError;
-use workload::apis::Error as WorkloadError;
+//use edgelet_http::Error as EdgeletHttpError;
+//use workload::apis::Error as WorkloadError;
 
-#[derive(Debug)]
-pub struct Error {
-    inner: Context<ErrorKind>,
-}
 
-#[derive(Debug, Fail)]
-pub enum ErrorKind {
-    #[fail(display = "Client error")]
-    Client(WorkloadError<serde_json::Value>),
-    #[fail(display = "Error connecting to endpoint")]
+#[derive(Debug, ThisError)]
+pub enum Error {
+    #[error("Error connecting to endpoint")]
     Http,
-    #[fail(display = "Hyper error")]
-    Hyper,
-    #[fail(display = "An IO error occurred.")]
-    Io,
-    #[fail(display = "Url parse error")]
-    Parse,
-    #[fail(display = "Serde error")]
-    Serde,
-}
-
-impl Fail for Error {
-    fn cause(&self) -> Option<&dyn Fail> {
-        self.inner.cause()
-    }
-
-    fn backtrace(&self) -> Option<&Backtrace> {
-        self.inner.backtrace()
-    }
-}
-
-impl Display for Error {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        Display::fmt(&self.inner, f)
-    }
-}
-
-impl From<ErrorKind> for Error {
-    fn from(kind: ErrorKind) -> Self {
-        Error {
-            inner: Context::new(kind),
-        }
-    }
-}
-
-impl From<Context<ErrorKind>> for Error {
-    fn from(inner: Context<ErrorKind>) -> Self {
-        Error { inner }
-    }
+    #[error("Hyper error")]
+    Hyper(
+        #[from]
+        #[source]
+        HyperError,
+      ),
```

Comment:
I typically don't see tests for these, but it might be nice to just confirm that the error messages look good to you and that there is no additional context that might be required. I only bring this up in case some of the errors from say Hyper are not super informative. Don't have to do this now in this PR, but something to consider :)